// @chiper-inc/sb-web-chiper/dist/SelectFields/MultipleDropdown/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MultipleDropdown = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const react_2 = require("react");
const deprecate_1 = require("../../Utils/deprecate");
const DropdownListItem_1 = require("../Dropdowns/DropdownListItem");
const DropdownWrapper_1 = require("../Dropdowns/DropdownWrapper");
/**
 * ⚠️ MultipleDropdown is deprecated. DO NOT USE! Use `SelectorGroup` instead.
 *
 * #### Interface:
 *
 * ```tsx
  export interface IDropdownItem {
    value: string;
    text: string;
    subItems?: IDropdownItem[];
  }
 * ```
 *
 * #### Usage:
 *
 * ```tsx
  import { MultipleDropdown } from "@chiper-inc/sb-web-chiper";

  const [values, setValues] = useState<string[]>([]);

  const handleClick = useCallback((itemValue: string) => {
    setValues((prev) => (prev.includes(itemValue)
      ? prev.filter((i) => i !== itemValue)
      : [...prev, itemValue]));
  }, []);

  <MultipleDropdown
    items={[
      {
        text: "Item 1",
        value: "123",
        icon: "FolderPlus",
        subItems: [
          { text: "Sub Item 1-2", value: "124" },
          { text: "Sub Item 1-3", value: "125" },
        ],
      },
      {
        text: "Item 2",
        value: "126"
      },
      {
        text:"Item 3",
        value: "127"
      },
      {
        text: "Item 4",
        value: "125",
        subItems: [
          { text: "Sub Item 4-1", value: "121", icon: "FolderDotted" },
          { text: "Sub Item 4-2", value: "1220", icon: "FolderLock" },
          { text: "Sub Item 4-3", value: "1203" },
          { text: "Sub Item 4-4", value: "1202" },

        ],
      },
  ]}
  selectedValues={values}
  onItemClick={handleClick}
/>
 * ```
 *
 * #### Demo:
 * @deprecated
 */
function MultipleDropdown({ items, onItemClick, selectedValues = [], width = "normal", maxHeight = "460px", }) {
    (0, deprecate_1.deprecate)({
        message: "MultipleDropdown is deprecated. Use SelectorGroup instead.",
    });
    const isSelected = (0, react_2.useCallback)((itemValue) => (selectedValues.includes(itemValue)), [selectedValues]);
    return ((0, jsx_runtime_1.jsx)(DropdownWrapper_1.DropdownWrapper, Object.assign({ width: width, maxHeight: maxHeight, "data-testid": "chiper-multiple-dropdown" }, { children: items.map((item, i) => ((0, react_1.createElement)(DropdownListItem_1.DropdownListItem, Object.assign({}, item, { type: "checkbox", key: `dropdown-item-${item.text + i}`, isSelected: isSelected, onClick: onItemClick })))) })));
}
exports.MultipleDropdown = MultipleDropdown;
