// @chiper-inc/sb-web-chiper/dist/Cards/MiniBrandCard/index.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MiniBrandCard = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const DEFAULT_IMAGE = "https://d221xxk5mfaxk5.cloudfront.net/app/no-image-ryahXCwGV-L.jpg";
/**
 * A `MiniBrandCard` is a representation of the `BrandCard` with fewer elements.
 * It's only to exemplify how it would be
 *
 * #### Usage:
 *
 * ```tsx
  import { MiniBrandCard } from "@chiper-inc/sb-web-chiper";

  <MiniBrandCard
    id="1045"
    brandName="Coca Cola"
    className="border border-greyscale-gray25"
    image="https://d221xxk5mfaxk5.cloudfront.net/app/cocacola_150-ry4-QI9l5-M.png"
    products={[
      {
        id: '7702535011805',
        name: 'Gaseosa - Coca Cola Original - Botella  1+1 L',
        image: 'https://d221xxk5mfaxk5.cloudfront.net/app/gaseosa-coca-cola-x-15-litros-11743-1-rypZzM9g8-M.png',
        currentSymbol: '$',
        originalPrice: '1,000'
      },
      {
        id: '7702535005941',
        name: 'Coca Cola Original - Coca Cola - Botella pet 1500 ml',
        image: 'https://d221xxk5mfaxk5.cloudfront.net/app/gaseosa-sabor-original-toronja-x-3000ml-coca-colaquatro-botella-pet-6000-ml-r1Lhy4xvr-M.png',
        currentSymbol: '$',
        originalPrice: '1,000'
      }
    ]}
  />
 * ```
 *
 * #### Demo:
 */
function MiniBrandCard(_a) {
    var _b, _c, _d, _e, _f, _g, _h;
    var { id, image, brandName, products = [{ id: "1" }, { id: "2" }] } = _a, rest = __rest(_a, ["id", "image", "brandName", "products"]);
    return ((0, jsx_runtime_1.jsx)("div", Object.assign({}, rest, { children: (0, jsx_runtime_1.jsxs)("div", Object.assign({ "data-testid": `mini-card-brand-${id}`, id: `mini-card-brand-${id}`, style: { aspectRatio: "0.95" }, className: `
          sbw-flex
          sbw-w-full
          sbw-max-w-sm
          sbw-flex-row
          sbw-gap-2
          sbw-overflow-hidden
          sbw-bg-white
          sbw-p-2
        ` }, { children: [(0, jsx_runtime_1.jsxs)("div", Object.assign({ className: "sbw-flex sbw-w-1/3 sbw-flex-col sbw-gap-1" }, { children: [(0, jsx_runtime_1.jsx)("figure", Object.assign({ "data-testid": `brand-image-${id}`, style: { aspectRatio: "10 / 10" }, className: `
              sbw-box-border
              sbw-flex
              sbw-items-center
              sbw-justify-center
              sbw-overflow-hidden
              sbw-rounded
              sbw-border
              sbw-border-greyscale-gray25
              ${image ? "sbw-bg-white" : "sbw-bg-greyscale-gray25"}
            ` }, { children: image && ((0, jsx_runtime_1.jsx)("img", { className: "sbw-object-fill", src: image || DEFAULT_IMAGE, alt: brandName, loading: "lazy", onError: ({ currentTarget }) => {
                                    currentTarget.onerror = null;
                                    currentTarget.src = DEFAULT_IMAGE;
                                } })) })), products.slice(0, 2).map((product) => ((0, jsx_runtime_1.jsx)("figure", Object.assign({ style: { aspectRatio: "10 / 10" }, className: `
                sbw-box-border
                sbw-flex
                sbw-items-center
                sbw-justify-center
                sbw-overflow-hidden
                sbw-rounded
                sbw-border
                sbw-border-greyscale-gray25
                ${(product === null || product === void 0 ? void 0 : product.image) ? "sbw-bg-white" : "sbw-bg-greyscale-gray25"}
              ` }, { children: (product === null || product === void 0 ? void 0 : product.image) && ((0, jsx_runtime_1.jsx)("img", { className: "sbw-object-fill", src: product.image || DEFAULT_IMAGE, alt: product.name, loading: "lazy", onError: ({ currentTarget }) => {
                                    currentTarget.onerror = null;
                                    currentTarget.src = DEFAULT_IMAGE;
                                } })) }), product.id)))] })), (0, jsx_runtime_1.jsxs)("div", Object.assign({ className: "sbw-w-2/3" }, { children: [(0, jsx_runtime_1.jsx)("figure", Object.assign({ style: { aspectRatio: "10 / 10" }, className: `
              sbw-flex
              sbw-items-center
              sbw-justify-center
              ${((_b = products === null || products === void 0 ? void 0 : products[0]) === null || _b === void 0 ? void 0 : _b.image) ? "sbw-bg-white" : "sbw-bg-greyscale-gray25"}
            ` }, { children: ((_c = products === null || products === void 0 ? void 0 : products[0]) === null || _c === void 0 ? void 0 : _c.image) && ((0, jsx_runtime_1.jsx)("img", { className: "sbw-object-fill", src: ((_d = products === null || products === void 0 ? void 0 : products[0]) === null || _d === void 0 ? void 0 : _d.image) || DEFAULT_IMAGE, alt: (_e = products === null || products === void 0 ? void 0 : products[0]) === null || _e === void 0 ? void 0 : _e.name, loading: "lazy", onError: ({ currentTarget }) => {
                                    currentTarget.onerror = null;
                                    currentTarget.src = DEFAULT_IMAGE;
                                } })) })), (0, jsx_runtime_1.jsxs)("div", Object.assign({ className: "\n              sbw-mt-1\n              sbw-text-left\n              sbw-text-xs\n              sbw-text-primary-darkBlueGray\n            " }, { children: [(0, jsx_runtime_1.jsxs)("p", Object.assign({ className: "\n                sbw-mt-0\n                sbw-leading-tight\n                sbw-line-clamp-1\n              " }, { children: [(_f = products === null || products === void 0 ? void 0 : products[0]) === null || _f === void 0 ? void 0 : _f.currentSymbol, " ", (_g = products === null || products === void 0 ? void 0 : products[0]) === null || _g === void 0 ? void 0 : _g.originalPrice] })), (0, jsx_runtime_1.jsx)("p", Object.assign({ className: "sbw-mt-0 sbw-leading-tight sbw-line-clamp-2" }, { children: (_h = products === null || products === void 0 ? void 0 : products[0]) === null || _h === void 0 ? void 0 : _h.name }))] }))] }))] })) })));
}
exports.MiniBrandCard = MiniBrandCard;
