// @chiper-inc/sb-web-chiper/dist/TextFields/SearchField/index.js
"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchField = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const helpers_1 = require("./helpers");
const Icon_1 = require("../../Theme/Icon");
const DEFAULT_IMAGE = "https://d221xxk5mfaxk5.cloudfront.net/app/no-image-ryahXCwGV-L.jpg";
/**
 * `SearchField` allows users to search for information (product) within
 * a specified data set. It provides a customizable search input field
 * with optional search results dropdown or list, as well as additional props
 *
 * #### Usage:
 *
 * ```tsx
  import { useState } from "react";
  import { SearchField } from "@chiper-inc/sb-web-chiper";
  import { SearchResult } from 'sb-web-chiper/dist/Form/SearchField/interface'

  const products = [
    {
      id: 1,
      title: "Bebida Hidratante Mandarina Gatorade",
      sku: "0987656432183",
      image: "https://supermercadoacuario.com.ar/app/files/company_35/products/65035_7792170042005.jpg",
      description: "500ml",
    },
    {
      id: 2,
      title: "Coca Col",
      sku: "098765632181",
      image: "https://coca-colafemsa.com/wp-content/uploads/2019/11/3-1.png",
      description: "500ml",
    },
    {
      id: 3,
      title: "Pepsi Cola",
      sku: "0987656432123",
      image: "https://d3ugyf2ht6aenh.cloudfront.net/stores/001/611/627/products/1500-regular-011-17aae1c17219bb3b8316179114560247-640-0.png",
      description: "500ml",
    },
  ];

  const [searchInput, setSearchInput] = useState("");
  const [items, setItems] = useState<SearchResult[]>([]);

  // Simple
  <SearchField
    placeholder="Ingrese el producto"
    items={products}
    value={searchInput}
    onChange={setSearchInput}
  />

  // Or With Result Click
  <SearchField
    placeholder="Ingrese el producto"
    items={products}
    value={searchInput}
    onChange={setSearchInput}
    onResultClick={(result) => {
      setSearchInput("");
      setItems((prevResult) => ([
        ...prevResult,
        {
          id: result.id,
          title: result.title,
          image: result.image,
          sku: result.sku,
          description: result.description,
        },
      ]));
    }}
  />
 * ```
 *
 * #### Demo:
 */
function SearchField(_a) {
    var _b;
    var { isList, items, placeholder, value, onChange, onClearResult, onResultClick, maxHeight, hasDebounce = false, disabled = false, className = "", smallHeight = false } = _a, props = __rest(_a, ["isList", "items", "placeholder", "value", "onChange", "onClearResult", "onResultClick", "maxHeight", "hasDebounce", "disabled", "className", "smallHeight"]);
    const inputSearchRef = (0, react_1.useRef)(null);
    const debounceFn = (0, react_1.useRef)({});
    const [results, setResults] = (0, react_1.useState)([]);
    const [isListOpen, setIsOpenList] = (0, react_1.useState)(false);
    const eraseField = (0, react_1.useCallback)(() => {
        var _a;
        setIsOpenList(false);
        if ((_a = inputSearchRef === null || inputSearchRef === void 0 ? void 0 : inputSearchRef.current) === null || _a === void 0 ? void 0 : _a.value) {
            inputSearchRef.current.value = "";
        }
        if (!isList && onClearResult) {
            setResults([]);
            onClearResult();
        }
    }, [isList, onClearResult]);
    const handleInputChange = (event) => {
        const newQuery = event.target.value;
        if (newQuery === "") {
            eraseField();
        }
        else {
            if (debounceFn.current) {
                clearTimeout(debounceFn.current);
            }
            debounceFn.current = setTimeout(() => {
                onChange(newQuery);
            }, (hasDebounce) ? 700 : 0);
        }
    };
    const handleDropdownClick = (product) => {
        eraseField();
        if (onResultClick) {
            onResultClick(product);
        }
    };
    const handleEscapeKey = (0, react_1.useCallback)((event) => {
        if (event.code === "Escape") {
            eraseField();
        }
    }, [eraseField]);
    (0, react_1.useEffect)(() => {
        var _a;
        if (items.length > 0 && ((_a = inputSearchRef === null || inputSearchRef === void 0 ? void 0 : inputSearchRef.current) === null || _a === void 0 ? void 0 : _a.value)) {
            const suggestions = (0, helpers_1.suggestResults)(inputSearchRef.current.value, items);
            const findSuggestions = suggestions.length > 0 ? suggestions : items;
            setResults(findSuggestions);
            setIsOpenList(true);
        }
        else {
            setIsOpenList(false);
            setResults([]);
        }
    }, [items, items.length]);
    (0, react_1.useEffect)(() => {
        document.addEventListener("keydown", handleEscapeKey);
        return () => {
            document.removeEventListener("keydown", handleEscapeKey);
        };
    }, [handleEscapeKey]);
    return ((0, jsx_runtime_1.jsxs)("div", Object.assign({}, props, { className: `
        sbw-relative
        ${className}
      ` }, { children: [(0, jsx_runtime_1.jsxs)("div", Object.assign({ className: `
          sbw-flex
          sbw-w-full
          sbw-items-center
          sbw-border
          sbw-border-greyscale-gray60
          sbw-px-4
          ${isListOpen ? "sbw-rounded-t-md sbw-border-b-0" : "sbw-rounded-md"}
          ${disabled ? "sbw-bg-greyscale-gray60" : "sbw-bg-primary-white"}
        ` }, { children: [(0, jsx_runtime_1.jsx)("div", Object.assign({ className: disabled ? "sbw-text-greyscale-gray200" : "sbw-text-primary-yankeesBlue" }, { children: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "MagnifyingGlass", size: 14 }) })), (0, jsx_runtime_1.jsx)("input", { type: "text", ref: inputSearchRef, "data-testid": "searchfield-input", placeholder: placeholder, defaultValue: value, onChange: handleInputChange, disabled: disabled, className: `
            sbw-w-full
            sbw-pl-2
            sbw-pr-4
            sbw-text-base
            focus:sbw-border-transparent
            focus:sbw-outline-none
            ${smallHeight ? "sbw-py-3" : "sbw-py-[14px]"}
            ${disabled
                            ? "sbw-bg-greyscale-gray60 sbw-text-greyscale-gray200 placeholder:sbw-text-greyscale-gray200"
                            : "sbw-bg-primary-white sbw-text-primary-yankeesBlue placeholder:sbw-text-primary-yankeesBlue"}
          ` }), isList ? ((0, jsx_runtime_1.jsx)("button", Object.assign({ type: "button", "data-testid": "search-btn-open", disabled: disabled, onClick: () => {
                            if (results.length > 0) {
                                setIsOpenList((prevOpenList) => !prevOpenList);
                            }
                        }, className: disabled
                            ? "sbw-text-greyscale-gray200"
                            : `sbw-cursor-pointer
                sbw-text-primary-yankeesBlue
                  focus-visible:sbw-outline
                  focus-visible:sbw-outline-2
                focus-visible:sbw-outline-secondary-brightYellow
                ` }, { children: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: isListOpen ? "CaretUp" : "CaretDown", size: 14 }) }))) : ((0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, { children: [(value && ((_b = inputSearchRef === null || inputSearchRef === void 0 ? void 0 : inputSearchRef.current) === null || _b === void 0 ? void 0 : _b.value)) && ((0, jsx_runtime_1.jsx)("button", Object.assign({ type: "button", "data-testid": "search-btn-close", disabled: disabled, onClick: eraseField, className: disabled
                                    ? "sbw-text-greyscale-gray200"
                                    : `sbw-cursor-pointer
                      sbw-text-primary-yankeesBlue
                      focus-visible:sbw-outline
                      focus-visible:sbw-outline-2
                    focus-visible:sbw-outline-secondary-brightYellow
                    ` }, { children: (0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "X", size: 14 }) }))), (0, jsx_runtime_1.jsx)("div", {})] }))] })), isListOpen && ((0, jsx_runtime_1.jsxs)("div", Object.assign({ "data-testid": "searchfield-results", style: { maxHeight }, className: `
            sbw-absolute
            sbw-left-0
            sbw-top-full
            sbw-z-50
            sbw-w-full
            sbw-overflow-x-hidden
            sbw-rounded-b-md
            sbw-border
            sbw-border-t-0
            sbw-border-greyscale-gray60
            sbw-bg-white
            ${maxHeight ? "sbw-overflow-y-scroll" : "sbw-overflow-y-auto"}
          ` }, { children: [(0, jsx_runtime_1.jsx)("hr", { className: "\n              sbw-mx-4\n              sbw-border\n              sbw-border-b-0\n              sbw-border-t-greyscale-gray25\n            " }), results.map((result, index) => ((0, jsx_runtime_1.jsx)("button", Object.assign({ type: "button", "data-testid": `searchfield-result-item-${index}`, onClick: () => handleDropdownClick(result), className: "\n                sbw-flex\n                sbw-w-full\n                sbw-cursor-pointer\n                sbw-border-white\n                sbw-px-4\n                sbw-text-left\n                last:sbw-rounded-b-md\n                hover:sbw-bg-greyscale-gray5\n                focus-visible:sbw-bg-greyscale-gray5\n                focus-visible:sbw-outline\n                focus-visible:sbw-outline-2\n                focus-visible:-sbw-outline-offset-2\n              focus-visible:sbw-outline-secondary-brightYellow\n              " }, { children: !isList ? ((0, jsx_runtime_1.jsxs)("div", Object.assign({ className: "sbw-flex" }, { children: [(result === null || result === void 0 ? void 0 : result.image) && ((0, jsx_runtime_1.jsx)("div", Object.assign({ className: "sbw-mr-2 sbw-flex sbw-max-w-12 sbw-items-center sbw-justify-center sbw-overflow-hidden" }, { children: (0, jsx_runtime_1.jsx)("img", { src: (result === null || result === void 0 ? void 0 : result.image) || DEFAULT_IMAGE, alt: result === null || result === void 0 ? void 0 : result.title, className: "sbw-h-10 sbw-object-contain", onError: ({ currentTarget }) => {
                                            currentTarget.onerror = null;
                                            currentTarget.src = DEFAULT_IMAGE;
                                        } }) }))), (0, jsx_runtime_1.jsxs)("div", Object.assign({ className: "sbw-my-2" }, { children: [(0, jsx_runtime_1.jsx)("div", { children: (0, helpers_1.boldMatchingText)(result.title, value) }), (result === null || result === void 0 ? void 0 : result.description) && ((0, jsx_runtime_1.jsx)("div", { children: result.description })), (result === null || result === void 0 ? void 0 : result.sku) && ((0, jsx_runtime_1.jsxs)("div", Object.assign({ className: "sbw-flex sbw-items-center" }, { children: [(0, jsx_runtime_1.jsx)(Icon_1.Icon, { name: "Barcode", size: 12 }), (0, jsx_runtime_1.jsx)("span", Object.assign({ className: "sbw-ml-1" }, { children: result.sku }))] })))] }))] }))) : ((0, jsx_runtime_1.jsx)("p", Object.assign({ className: "\n                    sbw-my-1\n                    sbw-text-primary-yankeesBlue\n                  " }, { children: isListOpen
                                ? result.title
                                : (0, helpers_1.boldMatchingText)(result.title, value) }))) }), result.id)))] })))] })));
}
exports.SearchField = SearchField;
