// @chiper-inc/sb-web-chiper/dist/Buttons/Fab/props.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.fabSizes = exports.fabSkins = void 0;
const Colors_1 = __importDefault(require("../../Theme/Colors"));
exports.fabSkins = {
    normal: {
        background: Colors_1.default.primary.green,
        border: Colors_1.default.primary.green,
        color: Colors_1.default.primary.white,
    },
    disabled: {
        background: Colors_1.default.greyscale.gray200,
        border: Colors_1.default.greyscale.gray200,
        color: Colors_1.default.primary.white,
    },
    outline: {
        background: Colors_1.default.primary.white,
        border: Colors_1.default.greyscale.gray60,
        color: Colors_1.default.primary.darkBlueGray,
    },
};
exports.fabSizes = {
    small: "sbw-w-[24px] sbw-h-[24px]",
    medium: "sbw-w-[32px] sbw-h-[32px]",
    large: "sbw-w-10 sbw-h-10",
};
