// @chiper-inc/sb-web-chiper/dist/Buttons/ChiperButton/props.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.loaderType = exports.uiTypeStyles = exports.sizeStyles = void 0;
const Colors_1 = __importDefault(require("../../Theme/Colors"));
exports.sizeStyles = {
    small: {
        button: "sbw-h-[32px]",
        text: "sbw-text-xs",
        icon: 16,
    },
    medium: {
        button: "sbw-h-[40px]",
        text: "sbw-text-base",
        icon: 24,
    },
    large: {
        button: "sbw-h-[48px]",
        text: "sbw-text-base",
        icon: 24,
    },
};
exports.uiTypeStyles = {
    primary: {
        button: "sbw-bg-primary-green disabled:sbw-bg-greyscale-gray60",
        hovered: "hover:sbw-bg-secondary-japaneseLaurel",
        text: "sbw-text-primary-white",
        iconColor: Colors_1.default.primary.white,
        iconClass: "",
    },
    secondary: {
        button: "sbw-bg-tertiary-honeydew disabled:sbw-bg-greyscale-gray60",
        hovered: "",
        text: "sbw-text-primary-green group-hover:sbw-text-secondary-japaneseLaurel group-disabled:sbw-text-primary-white",
        iconColor: Colors_1.default.primary.green,
        iconClass: "group-hover:sbw-fill-secondary-japaneseLaurel group-disabled:sbw-fill-primary-white",
    },
    outlined: {
        button: "sbw-bg-primary-white sbw-border-greyscale-gray60 disabled:sbw-bg-primary-white",
        hovered: "hover:sbw-bg-greyscale-gray25",
        text: "sbw-text-primary-darkBlueGray group-disabled:sbw-text-greyscale-gray60",
        iconColor: Colors_1.default.primary.darkBlueGray,
        iconClass: "group-disabled:sbw-fill-greyscale-gray60",
    },
    text: {
        button: "sbw-bg-transparent",
        hovered: "",
        text: "sbw-text-primary-green group-disabled:sbw-text-greyscale-gray60 group-hover:sbw-text-secondary-japaneseLaurel",
        iconColor: Colors_1.default.primary.green,
        iconClass: "group-disabled:sbw-fill-greyscale-gray60 group-hover:sbw-fill-secondary-japaneseLaurel",
    },
};
exports.loaderType = {
    primary: { type: "white" },
    secondary: { type: "green", insideBackgroundClass: "sbw-bg-tertiary-honeydew" },
    outlined: { type: "gray" },
    text: { type: "green" },
};
