// @chiper-inc/sb-web-chiper/dist/Theme/Colors/index.js
"use strict";
module.exports = {
    primary: {
        transparent: "transparent",
        yankeesBlue: "#2B2845",
        green: "#00B56B",
        chiperBrandRed: "#FF0136",
        darkBlueGray: "#666A8F",
        ghostWhite: "#FAFBFC",
        white: "#FFFFFF",
    },
    secondary: {
        bleuDeFrance: "#3584E0",
        blueJeans: "#56B0F3",
        brightYellow: "#FF9F26",
        philippineGreen: "#01834E",
        diferente: "#31549E",
        positiva: "#EF909B",
        visionaria: "#F4AA1C",
        cercana: "#5EBD78",
        creativa: "#7255A2",
        sonadora: "#66A0D6",
        /** @deprecated use japaneseLaurel */
        productiva: "#37723D",
        extraordinaria: "#FFEDD4",
        transformadora: "#A12822",
        green10: "#E5F8F0",
        japaneseLaurel: "#37723D",
    },
    greyscale: {
        gray200: "#A2A6BD",
        gray300: "#8084A2",
        gray100: "#C6C9D8",
        gray60: "#DDDFE8",
        gray25: "#EBEDF1",
        gray5: "#F9F9FA",
    },
    tertiary: {
        honeydew: "#E5F8F0",
        mistyRose: "#FFE6EB",
        cosmicLatte: "#FFF5E9",
        aliceBlue: "#EBF3FC",
    },
};
