// @chiper-inc/sb-web-chiper/dist/Feedback/Tooltip/TooltipArrow/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TooltipArrow = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const arrowClasses = {
    down: {
        position: `
      sbw-border-x-[6.75px]
      sbw-border-x-transparent
      sbw-border-b-[6.75px]
      sbw-bottom-full
      sbw-left-0
      sbw-right-0
    `,
        dark: "sbw-border-b-primary-yankeesBlue",
        light: "sbw-border-b-white",
    },
    up: {
        position: `
      sbw-border-x-[6.75px]
      sbw-border-x-transparent
      sbw-border-t-[6.75px]
      sbw-top-full
      sbw-left-0
      sbw-right-0
    `,
        dark: "sbw-border-t-primary-yankeesBlue",
        light: "sbw-border-t-white",
    },
    left: {
        position: `
      sbw-border-y-[6.75px]
      sbw-border-y-transparent
      sbw-border-l-[6.75px]
      sbw-left-full
      sbw-bottom-0
      sbw-top-0
    `,
        dark: "sbw-border-l-primary-yankeesBlue",
        light: "sbw-border-l-white",
    },
    right: {
        position: `
      sbw-border-y-[6.75px]
      sbw-border-y-transparent
      sbw-border-r-[6.75px]
      sbw-right-full
      sbw-bottom-0
      sbw-top-0
    `,
        dark: "sbw-border-r-primary-yankeesBlue",
        light: "sbw-border-r-white",
    },
};
function TooltipArrow({ position, modeDark, }) {
    const positionClass = arrowClasses[position].position;
    const colorClass = arrowClasses[position][modeDark ? "dark" : "light"];
    return ((0, jsx_runtime_1.jsx)("div", { className: `sbw-absolute sbw-m-auto sbw-h-0 sbw-w-0 ${positionClass} ${colorClass}` }));
}
exports.TooltipArrow = TooltipArrow;
