// @chiper-inc/sb-web-chiper/dist/Utils/useEventsQueue/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.wrapPromise = exports.isBeforeNow = exports.getNextExecution = exports.getNow = void 0;
/* eslint-disable no-promise-executor-return */
const dayjs_1 = __importDefault(require("dayjs"));
const getNow = () => (0, dayjs_1.default)().toISOString();
exports.getNow = getNow;
const getNextExecution = (ms) => (0, dayjs_1.default)().add(ms, "ms").toISOString();
exports.getNextExecution = getNextExecution;
const isBeforeNow = (date) => (0, dayjs_1.default)(date).isBefore((0, dayjs_1.default)());
exports.isBeforeNow = isBeforeNow;
const wrapPromise = (fn, onSucces, onError) => {
    const promise = new Promise((r) => r(fn()));
    function OnSucces(response) {
        if (onSucces) {
            onSucces(response);
        }
        return null;
    }
    function OnError(e) {
        if (onError) {
            onError(e);
        }
        return null;
    }
    return promise
        .then((response) => OnSucces(response))
        .catch((e) => OnError(e));
};
exports.wrapPromise = wrapPromise;
